app/bundles/CampaignBundle/Tests/Controller/CampaignTableStatsControllerTest.php

```diff
@@ -0,0 +1,235 @@
+use Mautic\CampaignBundle\Entity\Campaign;
+use Mautic\CampaignBundle\Entity\Event;
+use Mautic\CampaignBundle\Model\CampaignModel;
+use Mautic\CoreBundle\Helper\ExportHelper;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\CoreBundle\Translation\Translator;
+use Mautic\EmailBundle\Entity\Email;
+use PHPUnit\Framework\MockObject\MockObject;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class CampaignTableStatsControllerTest extends MauticMysqlTestCase
+{
+    private MockObject $campaignModelMock;
+
+    private CampaignTableStatsController $controller;
+
+    private MockObject $exportHelper;
+
+    private MockObject $translator;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $corePermissionsMock = $this->createMock(CorePermissions::class);
+        $corePermissionsMock->method('hasEntityAccess')
+            ->willReturn(true);
+
+        $this->campaignModelMock       = $this->createMock(CampaignModel::class);
+        $this->exportHelper            = $this->createMock(ExportHelper::class);
+        $this->translator              = $this->createMock(Translator::class);
+        $this->controller              = new CampaignTableStatsController(
+            $this->campaignModelMock,
+            $corePermissionsMock,
+            $this->exportHelper,
+            $this->translator
+        );
+    }
+
+    /**
+     * @return array<string, array<string, string>>
+     */
+    private function getStats(bool $addEmail = true): array
+    {
+        return $addEmail ? [
+            'Finland' => [
+                    'contacts'              => '14',
+                    'country'               => 'Finland',
+                    'sent_count'            => '14',
+                    'read_count'            => '4',
+                    'clicked_through_count' => '0',
+                ],
+            'Italy' => [
+                    'contacts'              => '5',
+                    'country'               => 'Italy',
+                    'sent_count'            => '5',
+                    'read_count'            => '5',
+                    'clicked_through_count' => '3',
+                ],
+        ] : [
+            'Finland' => [
+                    'contacts' => '14',
+                    'country'  => 'Finland',
+                ],
+            'Italy' => [
+                    'contacts' => '5',
+                    'country'  => 'Italy',
+                ],
+        ];
+    }
+
+    /**
+     * @throws OptimisticLockException
+     * @throws Exception
+     * @throws ORMException
+     */
+    public function testGetData(): void
+    {
+        $campaign = $this->createCampaignWithEmail();
+
+        $this->campaignModelMock->method('getCountryStats')
+            ->with($campaign)
+            ->willReturn($this->getStats());
```

Comment:
You are mixing here unit and functional tests. In the `createCamapignWithEmail` method you created a database objects, but later you used a mocked object and data. We should test the controllers using functional tests: create the database objects and call the endpoint as the user would.